test-specifications/token-service/tokenGrantKycTransaction.md

```diff
@@ -0,0 +1,141 @@
+| 11      | Grants KYC of a paused token to an account                                   | tokenId=<CREATED_TOKEN_ID>, accountId=<CREATED_ACCOUNT_ID>, commonTransactionParams.signers=[<CREATED_TOKEN_KYC_KEY>]       | The token KYC grant fails with an TOKEN_IS_PAUSED response code from the network.                 | N                 |
+| 12      | Grants KYC of a token to a frozen account                                    | tokenId=<CREATED_TOKEN_ID>, accountId=<CREATED_ACCOUNT_ID>, commonTransactionParams.signers=[<CREATED_TOKEN_KYC_KEY>]       | The token KYC grant fails with an ACCOUNT_FROZEN_FOR_TOKEN response code from the network.        | N                 |
+
+#### JSON Request Example
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 64362,
+  "method": "grantTokenKyc",
+  "params": {
+    "tokenId": "0.0.15432",
+    "accountId": "0.0.53848",
+    "commonTransactionParams": {
+      "signers": [
+        "302E020100300506032B657004220420DE6788D0A09F20DED806F446C02FB929D8CD8D17022374AFB3739A1D50BA72C8"
+      ]
+    }
+  }
+}
+```
+
+#### JSON Response Example
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 64362,
+  "result": {
+    "status": "SUCCESS"
+  }
+}
+```
+
+### **Account ID:**
+
+- The ID of the account to which to grant KYC.
+
+| Test no | Name                                                   | Input                                                                                                                 | Expected response                                                                     | Implemented (Y/N) |
+|---------|--------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------|-------------------|
+| 1       | Grants KYC of a token to an account that doesn't exist | tokenId=<CREATED_TOKEN_ID>, accountId="123.456.789", commonTransactionParams.signers=[<CREATED_TOKEN_KYC_KEY>]        | The token KYC grant fails with an INVALID_ACCOUNT_ID response code from the network.  | N                 |
+| 2       | Grants KYC of a token to an empty account ID           | tokenId=<CREATED_TOKEN_ID>, accountId="", commonTransactionParams.signers=[<CREATED_TOKEN_KYC_KEY>]                   | The token KYC grant fails with an SDK internal error.                                 | N                 |
+| 3       | Grants KYC of a token to an account with no account ID | tokenId=<CREATED_TOKEN_ID>, commonTransactionParams.signers=[<CREATED_TOKEN_KYC_KEY>]                                 | The token KYC grant fails with an INVALID_TOKEN_ID response code from the network.    | N                 |
```

Comment:
> Grants KYC of a token to an account with no account ID

I don't get that test scenario. Is the point to not set `accountId`? - if so, it would fail with `INVALID_ACCOUNT_ID`.

Reply:
You're right this is a typo, fixed

Reply:
An account can exist with an alias but no account ID (e.g. if auto-created from a transfer or smart contract action).  I assume this is testing that case.